pkg/operator/operator.go

```diff
@@ -469,14 +469,19 @@ func (o *Operator) sync(key string) error {
 		klog.Errorf("error occurred while setting status to in progress: %v", err)
 	}
 
-	taskName, err := tl.RunAll()
-	if err != nil {
-		klog.Infof("Updating ClusterOperator status to failed. Err: %v", err)
-		failedTaskReason := strings.Join(strings.Fields(taskName+"Failed"), "")
-		reportErr := o.client.StatusReporter().SetFailed(err, failedTaskReason)
-		if reportErr != nil {
-			klog.Errorf("error occurred while setting status to failed: %v", reportErr)
+	taskErrors := tl.RunAll()
+	if len(taskErrors) > 0 {
+		err := errors.New("Cluster monitoring update failed.")
+		for _, val := range taskErrors {
+			klog.Infof("Updating ClusterOperator status to failed. Err: %v", val.Err)
+			failedTaskReason := strings.Join(strings.Fields(val.Name+"Failed"), "")
+			err = errors.Wrap(err, failedTaskReason)
+			reportErr := o.client.StatusReporter().SetFailed(val.Err, failedTaskReason)
```

Comment:
if we get several errors, it will update the failed status condition several times and the last will win. We could just use the [list of tasks](https://github.com/openshift/cluster-monitoring-operator/blob/f4b1311c3928d8cb7ede47a7e6594a955ddd46be/pkg/operator/operator.go#L444-L464) and pick the first one that failed. Having said that the [`Message`](https://github.com/openshift/cluster-monitoring-operator/blob/f4b1311c3928d8cb7ede47a7e6594a955ddd46be/vendor/github.com/openshift/api/config/v1/types_cluster_operator.go#L136) field can concatenate the list of all errors.

Reply:
done